src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,10 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
Consider pushing the `methods[0].Arity == 0` check inside the `if` since the `else` currently calculates common return type and `methodContainer` which is unnecessary in the case of a single generic method.